pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -2165,35 +2206,23 @@ func (cc *XController) manageQueueJob(qj *arbv1.AppWrapper, podPhaseChanges bool
 				cc.quotaManager.Release(updateQj)
 			}
 
+            if derivedAwStatus == arbv1.AppWrapperStateCompleted {
+	           qj.Status.ItemCompletionStatus = genericItemsCompletionStatus
+			   qj.Status.State = derivedAwStatus
+            }
 			// Bugfix to eliminate performance problem of overloading the event queue.
 		} else if podPhaseChanges { // Continued bug fix
 			// Only update etcd if AW status has changed.  This can happen for periodic
 			// updates of pod phase counts done in caller of this function.
 			if err := cc.updateEtcd(qj, "manageQueueJob - podPhaseChanges"); err != nil {
 				klog.Errorf("[manageQueueJob] Error updating etc for AW job=%s Status=%+v err=%+v", qj.Name, qj.Status, err)
 			}
+			// Finish adding qj to Etcd for dispatch
+		//	return nil
```

Comment:
Likely need to leave `return nil` to not generate a side effect where the return is `return err`.

Reply:
The return nil is commented out in the original code (1.31). Not something I've changed.